### PR TITLE
[WIP] web interface for DSpot

### PR DESCRIPTION
**Pull request by Tailp**
Hi, this is a PR for the webinterface task . Per discussion today I need to get handle issue  #843 before this can really be deployed for actual use, since we need real data, but this is more or less done as a basic website for dspot :) .

Some facts: this web is build using MEAN stack , which stands for Mongodb, Express, AngularJS and NodeJS. Reason I am implementing this is because it has Mongodb in it since Repairnator also use Mongodb and this is just for a small website so no need to super flashy stuffs (yet) :) and also I know that you guys (Both on Dspot and Repairnator) like Java so basicly this stack is heavily focused on Javascript which is perfect for that reason . Other than that the html pages are ofcourse build with the classical html, javascript and css together with bootstrap 4 for styling. 

To use you need mongodb installed on your machine, preferably version 4.0 or above (since I am using it and It works for me and I can't guarantee for other version yet).

Install basic dependencies and mongodb dependecy.
```
  cd dspot-web
  npm install 
  npm install mongodb
```
To run 
```
  cd dspot-web
  DEBUG=myapp:* npm start
```
Then check out at [http://localhost:3000](http://localhost:3000) . Note that since you don't have any data it will look a bit empty :) .

### CODE REVIEW

**Comment by monperrus** [comment]
Cool progress! Could you add `dspot-web/README.md` to explain what it does and how it deploy it?

**Comment by Tailp** [comment]
> 
> 
> Cool progress! Could you add `dspot-web/README.md` to explain what it does and how it deploy it?

Yes I plan to do that after I am done with the Mongodb feature :) , since I also might need to change the current webinterface to fit with this new feature for Dspot.

**Comment by monperrus** [comment]
add README.md to explain how to deploy it

**Comment by monperrus** [comment]
Good progress!

First comments:

* remove Dockerfile
* add screenshots and explanation in README.md
* remove from the UI the selector choice
* remove from the UI the branch choice (always master or develop)

**Comment by Tailp** [comment]
resolved :). 
Travis: Only failed check style unrelated to this PR.

File: dspot-web/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+```
+cd dspot-web
+npm install --no-optional
+npm run-script start
+```
+
+## Screenshots
+
+![Pic 1](https://github.com/Tailp/dspot/blob/web/dspot-web/screenshots/pic1.png)
```
Comment (monperrus) [inline dspot-web/README.md]: Thanks for the screenshots. Could you add one paragraph per screenshot to explain what they show?

**Comment by Tailp** [comment]
@monperrus sorry for some duplicated commits. Server was laggy, I thought that my commits failed :). Resolved. Awaiting for new comments.

File: dspot-web/README.md (hunk 1 above)
Comment (monperrus) [reply dspot-web/README.md]: merge pic1 and pic2 they are the same.

File: dspot-web/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+### Front page
+![Pic 1](https://github.com/Tailp/dspot/blob/web/dspot-web/screenshots/pic1.png)
+![Pic 2](https://github.com/Tailp/dspot/blob/web/dspot-web/screenshots/pic2.png)
+### All amplified repos page
+![Pic 3](https://github.com/Tailp/dspot/blob/web/dspot-web/screenshots/pic3.png)
+### Demo submission
+
+A repo has been submitted
+
+![Pic 4](https://github.com/Tailp/dspot/blob/web/dspot-web/screenshots/pic4.png)
```
Comment (monperrus) [inline dspot-web/README.md]: pic 4: the "ack screen" should not contain the form and the list of recent repositories

File: dspot-web/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+Display pending status on repos page
+
+![Pic 5](https://github.com/Tailp/dspot/blob/web/dspot-web/screenshots/pic5.png)
```
Comment (monperrus) [inline dspot-web/README.md]: Pic 5 should replace Pic 3, no need to have both

File: dspot-web/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+When done, specific repo page display details
+
+![Pic 6](https://github.com/Tailp/dspot/blob/web/dspot-web/screenshots/pic6.png)
```
Comment (monperrus) [inline dspot-web/README.md]: Merge Pic 6 and Pic 7.

File: dspot-web/controllers/pages/index.html [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+                                    <strong>{{response}}</strong>
+                                </div>
+                            </form>
+                        </div>
+                    </div>
+                </div>
+                <div class="col-sm-4">
+                    <h2>Important!</h2>
+                    <div class="entry-content">
+                        <p>&#9642; A config file 'dspot.properties' at maven project root is highly recommended (check out our Git repo for more details)</p>
```
Comment (monperrus) [inline dspot-web/controllers/pages/index.html]:
```suggestion
                        <p>&#9642; If a config file 'dspot.properties' is present at the repository root, it is taken into account (<a href ="TODO">documentation</a>)</p>
```

File: dspot-web/controllers/pages/index.html [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+                        <p>&#9642; If no config file, auto-config is provided only for default non-multimodule maven project structure </p>
```
Comment (monperrus) [inline dspot-web/controllers/pages/index.html]:
```suggestion
                        <p>&#9642; Auto-config is provided only for single module maven projects ( non-multimodule) </p>
```

File: dspot-web/README.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
+# Introduction
+
+This is a prototype web interface of Dspot, inspired from [CommitGuru](http://commit.guru/).
+
```
Comment (monperrus) [inline dspot-web/README.md]: Add the URL to the prototype (eg https://dspot-web-prototype.kth-assert.net)

**Comment by monperrus** [comment]
Thanks @Tailp for the progress.

File: dspot-web/README.md (hunk 2 above)
Comment (Tailp) [reply dspot-web/README.md]: Hi , I have some troubles understanding this. Could it be that you mean if I submit a repo then both the form and a list dissappear ?

**Comment by monperrus** [comment]
yes, it's a new screen, quite empty, with a nice picture (animation?), saying in a big font

"We'll come back to you over email. The DSpot team.

(you can now safely close this window)
"

**Comment by Tailp** [comment]
Hi :), I believe a modal would fit your preference perfectly. It's an overlay window upon the current page after clicking on the submit button. WDYT ? :)

**Comment by Tailp** [comment]
@monperrus  hi :) , Issues resolved more or less. Here is a [link](35.246.191.225:3000) to dspot app. I didn't set up a proper url like you proposed yet since I am deploying this with my own cluster. @gluckzhang can do the DNS domain on KTH cluster  :).

**Comment by gluckzhang** [comment]
Thanks @Tailp for the progress! You could email me the ip address so that I will set up a domain name for your demo. If the project has been containerized, I could also help to deploy it into our k8s cluster.

**Comment by Tailp** [comment]
> Thanks @Tailp for the progress! You could email me the ip address so that I will set up a domain name for your demo. If the project has been containerized, I could also help to deploy it into our k8s cluster.

@gluckzhang thank you :) , I'll send you the two yaml files later, One for the webapp and the other for the pipeline. These can be done with the mongodb and activeMQ we already have previously for repairnator.

File: dspot-web/README.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+npm run-script start
+```
+
+## Screenshots
+
+### Front page
+
+The front page enables to submit a new repo.
+
+![Pic 1](https://github.com/Tailp/dspot/blob/web/dspot-web/screenshots/pic1.png)
```
Comment (monperrus) [inline dspot-web/README.md]: merging means a single screenshot with the complete window, not two concatenated windows :-)

Comment (Tailp) [reply dspot-web/README.md]: Ah, I see. I'll do that, but you'll have to zoom in later since I need to zoom out to capture the whole screen :).

**Comment by Tailp** [comment]
Hi @monperrus , it's resolved :) . Awaiting for new comments.

**Comment by monperrus** [comment]
LGTM @danglotb merge?

**Comment by monperrus** [comment]
Great milestone, thanks to all for the great work!

**Pull request merged**